extension/healthcheckextension/config.go

```diff
@@ -25,5 +25,10 @@ type Config struct {
 
 	// Port is the port used to publish the health check status.
 	// The default value is 13133.
+	// Deprecated: use Endpoint instead.
 	Port uint16 `mapstructure:"port"`
+
+	// Endpoint is the address and port used to publish the health check status.
+	// The default value is "localhost:13133".
+	Endpoint string `mapstructure:"endpoint"`
```

Comment:
Use `confignet`:

```
TcpAddr confignet.TcpAddr `mapstructure:",squash"`
```

Reply:
Fixed, I'd also suggest using `confignet.TCPAddr` instead of string in other extensions for the sake of consistency e.x: 
https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/pprofextension/config.go#L29

I can fix it next PR.